packages/beacon-node/src/chain/archiveStore/utils/updateBackfillRange.ts

```diff
@@ -0,0 +1,50 @@
+import {CheckpointWithHex} from "@lodestar/fork-choice";
+import {Logger} from "@lodestar/logger";
+import {IBeaconDb} from "../../../db/interface.js";
+import {IBeaconChain} from "../../interface.js";
+
+/**
+ * Backfill sync relies on verified connected ranges (which are represented as key,value
+ * with a verified jump from a key back to value). Since the node could have progressed
+ * ahead from, we need to save the forward progress of this node as another backfill
+ * range entry, that backfill sync will use to jump back if this node is restarted
+ * for any reason.
+ * The current backfill has its own backfill entry from anchor slot to last backfilled
+ * slot. And this would create the entry from the current finalized slot to the anchor
+ * slot.
+ */
+export async function updateBackfillRange(
```

Comment:
why was this moved in separate function? it's not clear from just looking at the diff in this PR

Reply:
Described in the motivation of the PR. 

> Make the archive store as slim possible with only publicly useful interface.

Reply:
> Make the archive store as slim possible with only publicly useful interface.

yes I read that but that misses the part why it even needs to be as slim as possible which for me would be the main rationale I would like to know to better understand they changes you are doing